objc/objc_runtime_darwin.go

```diff
@@ -544,7 +506,7 @@ func (i Ivar) Offset() uintptr {
 }
 
 // Protocol is a type that declares methods that can be implemented by any class.
-type Protocol uintptr
+type Protocol struct{}
```

Comment:
Is this really safe to change the type? There is no guarantee two pointers to an empty struct are different

https://go.dev/ref/spec#Size_and_alignment_guarantees

> Two distinct zero-size variables may have the same address in memory.

Reply:
Pointers to this struct are never created on the Go side. They should only be created by calling GetProtocol. In C this struct isn't empty but Go doesn't need to access any of its fields which is why I changed it to an empty struct so that users don't think to dereference it.

Reply:
Don't we have to distinguish Go's Protocol A and Go's Protocol B, for example? If so, why do we need this type?

Reply:
OK so we usually don't use the struct{} value itself but use pointers to structs, and we don't have to compare Protocol values directly. Is this correct?

Another possible proposal would be `[0]func()` to prevent comparing Protcol values directly.

Reply:
Yes we use pointers to structs. And yes we don't compare protocols directly there is a function protocol_isEqual that is used for that so perhaps that should be added too? 

I like making it explicit that it's not comparable so I'll use your suggestion